### DISCUSSION TITLE
CrystalNN for disordered structures

### DISCUSSION THREAD

**Discussion by SongjiaKong** [materialsproject/pymatgen#4046] [Q&A] [answered]
I am about to obtain the coordinates of the anion coordinated to the cation in the structure, thus I tried the CrystalNN, but when it meets disordered structure, it fails. I dont know is there any methods for disordered structures? I have already tried OrderDisorderedStructureTransformation and it did work, but the time was so long. 

By the way, the coordinate polyhedra even for disordered structures could be visualized in Vesta, I think the vesta principle for polyhedra identification may be helpful.

Thanks for help!

**Comment by janosh** [comment, marked as answer]
you could give [`SQSTransformation`](https://pymatgen.org/pymatgen.transformations.html#pymatgen.transformations.advanced_transformations.SQSTransformation) a try (but be careful to keep the `scaling` small or it'll take a long time as well)

**Reply by SongjiaKong** [reply]
By the way, I am not familiar with pymatgen, is there any functions to identify the connection of coordinate polyhedra?
